src/state/user/reducer.ts

```diff
@@ -27,6 +28,8 @@ export interface UserState {
   userDarkMode: boolean | null // the user's choice for dark mode or light mode
   matchesDarkMode: boolean // whether the dark mode media query matches
 
+  userLocale: string
```

Comment:
imo prefer `string | undefined` and treat undefined as falling back to detection

Reply:
`SupportedLocale` is an enum? I'm in favor of using an enum instead of string.